File: QuanLyPhongKham.Server/src/QuanLyPhongKham.Models/Entities/HOADON.cs
using MayNghien.Infrastructures.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuanLyPhongKham.Models.Entities
{
    public class HOADON : BaseEntity
    {
        [Key]
        public Guid MaHD { get; set; }
        public DateTime NgayThanhToan { get; set; }
        public decimal TongTien { get; set; }
        public string? TrangThaiThanhToan { get; set; }
        public string? MaLeTan { get; set; }

        [ForeignKey("PhieuKham")]
        public Guid? MaPK { get; set; }
        public PHIEUKHAM? PhieuKham { get; set; }
    }
}

File: QuanLyPhongKham.Client/src/models/HoaDon.js
/**
 * @typedef {Object} HoaDon
 * @property {string} maHD
 * @property {string|Date} ngayThanhToan
 * @property {number} tongTien
 * @property {string} trangThaiThanhToan
 * @property {string} maLeTan
 * @property {string} maPK
 * @property {string} tenBenhNhan
 */

export const createHoaDon = (
	maHD = "",
	ngayThanhToan = null,
	tongTien = 0,
	trangThaiThanhToan = "",
	maLeTan = "",
	maPK = "",
	tenBenhNhan = "",
) => ({
	maHD,
	ngayThanhToan,
	tongTien,
	trangThaiThanhToan,
	maLeTan,
	maPK,
	tenBenhNhan,
});

export const normalizeHoaDon = (item) => ({
	maHD: item?.maHD ?? item?.MaHD ?? "",
	ngayThanhToan: item?.ngayThanhToan ?? item?.NgayThanhToan ?? null,
	tongTien: Number(item?.tongTien ?? item?.TongTien ?? 0),
	trangThaiThanhToan:
		item?.trangThaiThanhToan ?? item?.TrangThaiThanhToan ?? "",
	maLeTan: item?.maLeTan ?? item?.MaLeTan ?? "",
	maPK: item?.maPK ?? item?.MaPK ?? "",
	tenBenhNhan: item?.tenBenhNhan ?? item?.TenBenhNhan ?? "",
});
